pp_hot.c

```diff
@@ -3979,6 +4135,7 @@ Perl_do_readline(pTHX)
         }
     }
     else {
+        /* XXX on RC builds, push on stack rather than moralise ? */
```

Comment:
"moralize" -> "mortalize" here and below.

Even if we don't push it to the value, it might be good to re-use the mortal stack entry, but I can see that being out of scope for this series of commits.

See #21654 for what led to this comment.  fea90cfbe1f reduced the CPU cost of such incremental tmps stack extensions, but we still do a lot of avoidable clean up of the mortals.